examples/single_active_consumer/single_active_consumer_super_stream.rs

```diff
@@ -0,0 +1,79 @@
+use futures::StreamExt;
+use rabbitmq_stream_client::error::StreamCreateError;
+use rabbitmq_stream_client::types::{
+    ByteCapacity, OffsetSpecification, ResponseCode, SuperStreamConsumer,
+};
+use std::collections::HashMap;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    use rabbitmq_stream_client::Environment;
+    let environment = Environment::builder().build().await?;
+    let message_count = 1000000;
+    let super_stream = "hello-rust-super-stream";
+
+    let create_response = environment
+        .stream_creator()
+        .max_length(ByteCapacity::GB(5))
+        .create_super_stream(super_stream, 3, None)
+        .await;
+
+    if let Err(e) = create_response {
+        if let StreamCreateError::Create { stream, status } = e {
+            match status {
+                // we can ignore this error because the stream already exists
+                ResponseCode::StreamAlreadyExists => {}
+                err => {
+                    println!("Error creating stream: {:?} {:?}", stream, err);
+                }
+            }
+        }
+    }
+    println!(
+        "Super stream consumer example, consuming messages from the super stream {}",
+        super_stream
+    );
+
+    let mut properties = HashMap::new();
+
+    properties.insert("single-active-consumer".to_string(), "true".to_string());
+    properties.insert("name".to_string(), "consumer-group-1".to_string());
+    properties.insert("super-stream".to_string(), "hello-rust-super-stream".to_string());
+
+    let mut super_stream_consumer: SuperStreamConsumer = environment
+        .super_stream_consumer()
+        .offset(OffsetSpecification::First)
+        .client_provided_name("my super stream consumer for hello rust")
+        /*We can decide a strategy to manage Offset specification in single active consumer based on is_active flag
+        By default if this clousure is not present the default strategy OffsetSpecification::NEXT will be set.*/
+        .consumer_update(move |active, message_context| {
+            println!("single active consumer: is active: {} on stream {}", active, message_context.get_stream());
+            OffsetSpecification::First
```

Comment:
We should make the example more complete. 
Instead of `OffsetSpecification::First` we should reload the offset from the server else it is not clear how it works.

Please have a look on:
https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/blob/main/docs/SuperStream/SuperStreamConsumer.cs#L47

The java client does the same